### ISSUE TITLE
Fichier inf3135-h2019-tp3.correction et test dans depot

### ISSUE THREAD

**Issue by ahmedsalia** [guyfrancoeur/INF3135_H2019_TP3#19]
Bonjour Pr @guyfrancoeur , 

Je me demandais pourquoi dans le fichier` inf3135-h2019-tp3.correction`, il n'y pas de cas qui testent les modification que nous avons apporter pour faire le TP3 (**Ex : test avec -d ou code de retour 7** ) 

Aussi, les fichiers contenus dans dans `tp3-data.zip`, ne contiennent qu'une seule ligne chaque. Serait- il possible d'avoir des tests pouvant nous aider à confirmer que le ce que nous faisons est bon ? 

Merci.

**Comment by guyfrancoeur** [comment]
Allo à tous,

Vous devez completer les tests par vous même, vous êtes capable (tous) !

Par contre, tous ceux et celles qui avez des tests et que vous aimeriez partager pour les gens qui ne sont pas capables, pas le temps ou pour tout autres raisons..  Je vous laisse le soin de nous dire que vous seriez prêt à partager vos travaux avec l'ensemble.  Je vais réfléchir si le cas se présente.

**Comment by leohersant** [comment]
Question sur le test no. 24 et 25

A)      1 90 0 test 2 exe            ./tp2 -c $(cat cp.txt) -i ~/depot/data2.in > res.out
B)      1 10 0 test 2 resultat       diff --brief -b res.out ~/depot/data2.res

La ligne A nous donne dans res.out le résultat suivant (car ordre DES par défaut)
137438691328
8589869056
33550336

Le contenu du fichier ~/depot/data2.res
33550336
8589869056
137438691328

La ligne B donne un échec, car, bien sûr, l'ordre n'est pas pareil... 

En fait qu'est-ce qu'on fait avec ce genre de situations?
Soit il faut mettre -d ASC dans la ligne A ou soit il faut changer le fichier data2.res

Des points bonis sont les bienvenus :)

**Comment by guyfrancoeur** [comment]
Allo,

J'ai regardé le contenu des fichiers data*.res présents dans tp3-data.zip.
+ Il y avais deux erreurs, bravo à tout ceux qui ont trouvés les erreurs;
+ Les fichiers sont corrigés dans le zip tp3-data.zip;

Comme mentionné par plusieurs le fichier tp2-data.zip contient une version (*.res) qui n'est pas dans l'ordre voulu.

**Comment by LauriGui** [comment]
PS: Le fichier `data2.res` n'est toujours pas valide, il y a une ligne vide, à la fin du fichier ;)

**Comment by dbrsseau** [comment]
@guyfrancoeur je ne comprends pas non plus la ligne 8 de votre fichier de correction. Vous vérifiez si le `clean` enlève les fichiers objets du tp2 alors que notre `make` génère ceux du `tp3`. Il aurait peut-être fallu faire comme votre ligne 12 avant la 8 ?

**Comment by LauriGui** [comment]
@dbrsseau,
Non, car il me semble qu'on souhaiterait que le `make` génère un lien symbolique `tp2`, comme souvent mentionné, dans d'autres `issues` ... Donc, on voudrait aussi, dans ce cas, que `make clean` soit ajusté, en conséquence...

**Comment by dbrsseau** [comment]
Il crée un lien symbolique de l'exécutable seulement (voir le test 12) et aussi, il le crée après le `clean` alors ça me semble un peu inutile de supprimer un fichier avant de l'avoir créé...

**Comment by LauriGui** [comment]
@dbrsseau,
Ce n'est pas que le fichier de tests qui serait « supposé » créer un lien symbolique...

**Comment by dbrsseau** [comment]
Eh bien, c'est pourtant ce qu'il fait dans le fichier de correction à la ligne 12...

**Comment by guyfrancoeur** [comment]
Allo,

La ligne vide dans le fichier data2.res est effacé.

Le zip est à jour!

**Comment by samisihem** [comment]
Bonjour,
j'ai remarqué que l'intervalle dans le fichier data2.in est de 10000 140000000000, ce qui donne comme résultat normalement :
137438691328 
8589869056 
33550336 
8128  
alors que le contenue du fichier data2.res est:
137438691328
8589869056
33550336
**il manque le nombre parfait :8128** ?
Merci.

**Comment by LauriGui** [comment]
@samisihem,
8128 < 10000 
😉

**Comment by samisihem** [comment]
@LauriGui,
Oui je viens de me rendre compte...et de me reveiller je crois :)
Merci.
